binary_transparency/firmware/cmd/ft_personality/internal/server.go

```diff
@@ -0,0 +1,115 @@
+// Package internal contains private implementation details for the FirmwareTransparency personality server.
+package internal
+
+import (
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"mime"
+	"mime/multipart"
+	"net/http"
+	"strings"
+
+	"github.com/golang/glog"
+	"github.com/google/trillian-examples/binary_transparency/firmware/api"
+)
+
+// Server is the core state & handler implementation of the FT personality.
+type Server struct {
+}
+
+// addFirmware handles requests to log new firmware images.
+// It expects a mime/multipart POST consisting of FirmwareMetadata json,
+// followed by a signature over those bytes.
+// TODO: store the actual firmware image in a CAS too.
+//
+// curl -i -X POST -H "Content-Type: multipart/mixed" -F 'json=@testdata/firmware_metadata.json;type=application/json' -F 'sig=@testdata/firmware_metadata.json.sig' localhost:8000/ft/v0/add_firmware
```

Comment:
Add a "example usage" either in the package comments or for this method.

Reply:
done